server/index.js

```diff
@@ -0,0 +1,106 @@
+const express = require('express');
+const path = require ('path');
+const models = require ('../database/models.js');
+const cookieParser = require('cookie-parser');
+const app = express();
+const port = 734;
+const options = {
+  maxAge: 2.592e+9,
+  httpOnly: true,
+}
+
+app.use(express.json())
+app.use(cookieParser())
+
+app.use('/',express.static(path.join(__dirname, '../client', 'public')))
+
+app.post('/api/addUser', (req, res) => {
+  models.saveUser(req.body.userName, req.body.password)
+  .then(
+    (newUser) => {
+      if (newUser) {
+        res.send(true)
+        console.log(`User: ${req.body.userName} Added`)
+      } else {
+        res.send(false)
+        console.log('Account Creation Failed')
+      }
+    }
+  )
+  .catch(err => console.log(err))
+  res.send('Dobby is a Good House Elf Sir')
+})
+// pass back userName and use it in useEffect query
+app.post('/api/login', (req, res) => {
+  models.login(req.body.userName, req.body.password)
+  .then((userName) => {
+    if (userName) {
+      res.cookie('session', `${userName}`, options)
+      res.send(true)
+      console.log('successful login')
+    } else {
+      res.send(false)
+      console.log('failed login')
+    }
+  })
+  .catch(err => console.log(err))
+})
+
+app.post('/api/updateLocations', (req, res) => {
+  if (req.body.locations) {
+  models.updateLocations(req.body.userName, req.body.currentLocation, req.body.locations)
```

Comment:
This seems like a security issue. Any account can update the locations of any other account. You shouldn't be reading in the userName from the request body, you should use the userName attached to the currently active session.